diff --git a/include/libopencm3/stm32/f3/st_usbfs.h b/include/libopencm3/stm32/f3/st_usbfs.h
index 7da73f2a..7b65d2b8 100644
--- a/include/libopencm3/stm32/f3/st_usbfs.h
+++ b/include/libopencm3/stm32/f3/st_usbfs.h
@@ -22,6 +22,6 @@
 #	error Do not include directly !
 #else
 
-#include <libopencm3/stm32/common/st_usbfs_v1.h>
+#include <libopencm3/stm32/common/st_usbfs_v2.h>
 
 #endif
diff --git a/lib/stm32/f3/Makefile b/lib/stm32/f3/Makefile
index 1c098e81..e8dec17d 100644
--- a/lib/stm32/f3/Makefile
+++ b/lib/stm32/f3/Makefile
@@ -51,7 +51,7 @@ OBJS		+= i2c_common_v2.o
 OBJS		+= spi_common_all.o spi_common_v2.o
 
 OBJS		+= usb.o usb_control.o usb_standard.o usb_msc.o
-OBJS		+= st_usbfs_core.o st_usbfs_v1.o
+OBJS		+= st_usbfs_core.o st_usbfs_v2.o
 
 VPATH += ../../usb:../:../../cm3:../common
 
